reposettings.py

```diff
@@ -142,15 +142,17 @@ def name():
         return "Branch protection settings hook"
 
     @staticmethod
-    def set(repo: Repository, config):
+    def set(repo: Repository.Repository, config):
         print(" Processing branch protection settings...")
 
         if 'branch-protection' not in config and 'branch-protection-overrides' not in config:
             print(" Nothing to do.")
             return
 
+        should_protect_default_branch = config.get('protect-default-branch')
+
         for branch in repo.get_branches():
-            if not branch.protected:
+            if not (branch.protected or (should_protect_default_branch and repo.default_branch == branch.name)):
```

Comment:
If entering here following the second part of the `if`, it is possible that we call `branch.edit_protection` when `branch.protected == false`. Is this allowed by the GH client?

Reply:
If I understood it correctly, a protected branch is just a branch with some kind of protection created (although the branch entity includes a boolean field to easily check if it is protected, that field is not editable and it gets its value from the protection field instead), therefore the only thing to do to make any branch protected is [adding some protection protection](https://docs.github.com/en/rest/branches/branch-protection#update-branch-protection) which is what is done by the `branch.edit_protection(...)` call.

Reply:
Some evidences to be sure:

Unprotected branch:
```
❯ curl -u $USER:$TOKEN \
  -H "Accept: application/vnd.github.v3+json" \
  https://api.github.com/repos/sigilioso/reposettings/branches/feature/support-protecting-default-branch/protection
{
  "message": "Branch not protected",
  "documentation_url": "https://docs.github.com/rest/reference/repos#get-branch-protection"
}
```

Then update its protection settings:

```
❯ curl -u $USER:$TOKEN -X PUT \
  -H "Accept: application/vnd.github.v3+json" \
  https://api.github.com/repos/sigilioso/reposettings/branches/feature/support-protecting-default-branch/protection \ 
  -d '{"enforce_admins": false, "required_pull_request_reviews": {"required_approving_review_count": 1}, "required_status_checks": null, "restrictions": null}'
{
  "url": "https://api.github.com/repos/sigilioso/reposettings/branches/feature/support-protecting-default-branch/protection",
  "required_pull_request_reviews": {
    "url": "https://api.github.com/repos/sigilioso/reposettings/branches/feature/support-protecting-default-branch/protection/required_pull_request_reviews",
    "dismiss_stale_reviews": false,
    "require_code_owner_reviews": false,
    "required_approving_review_count": 1
  },
  "required_signatures": {
    "url": "https://api.github.com/repos/sigilioso/reposettings/branches/feature/support-protecting-default-branch/protection/required_signatures",
    "enabled": false
  },
  "enforce_admins": {
    "url": "https://api.github.com/repos/sigilioso/reposettings/branches/feature/support-protecting-default-branch/protection/enforce_admins",
    "enabled": false
  },
  "required_linear_history": {
    "enabled": false
  },
  "allow_force_pushes": {
    "enabled": false
  },
  "allow_deletions": {
    "enabled": false
  },
  "block_creations": {
    "enabled": false
  },
  "required_conversation_resolution": {
    "enabled": false
  }
}
```